### PR TITLE
Improved unet and Dice Metric

### PR DESCRIPTION
**Pull request by mibaumgartner**
* Add normliazation to unet
* Dice metric

### CODE REVIEW

File: deliravision/torch/metrics.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import logging
+
+from deliravision.torch.utils.tensor_ops import make_onehot_npy
+
+
+logger = logging.getLogger(__file__)
+
+
+def compute_stat_score(pred, label, n):
```
Comment (justusschock) [inline deliravision/torch/metrics.py]: If this works on numpy, we should maybe have this in `vision_base`?

File: deliravision/torch/metrics.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    tp = ((cls_pred == n) * (label == n)).sum()
+    fp = ((cls_pred == n) * (label != n)).sum()
+    tn = ((cls_pred != n) * (label != n)).sum()
+    fn = ((cls_pred != n) * (label == n)).sum()
+
+    return tp, fp, tn, fn
+
+
+def dice_score(pred, label, bg=False, cls_logging=False,
```
Comment (justusschock) [inline deliravision/torch/metrics.py]: Also move this to `vision_base`?

File: deliravision/torch/metrics.py (hunk 1 above)
Comment (justusschock) [inline deliravision/torch/metrics.py]: If this works on numpy, we should maybe have this in `vision_base`?

File: deliravision/torch/metrics.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+from deliravision.torch.utils.tensor_ops import make_onehot_npy
+
+
+logger = logging.getLogger(__file__)
+
+
+def compute_stat_score(pred, label, n):
+    """
+    Compute metric statistics (true positive, false positive, true negative,
```
Comment (justusschock) [inline deliravision/torch/metrics.py]: maybe factor out the scores to own functions and call them here? Otherwise one cannot calc only one of them

File: deliravision/torch/models/backbones/unet.py [hunk 4]
```diff
@@ -289,18 +293,20 @@ def __init__(self, n_dim, in_channels, out_channels,
...
 
             def forward(self, from_down, from_up):
                 from_up = self.upconv(from_up)
                 if self.merge_mode == 'concat':
                     x = torch.cat((from_up, from_down), 1)
                 else:
                     x = from_up + from_down
-                x = F.relu(self.conv1(x))
-                x = F.relu(self.conv2(x))
+                x = F.relu(self.norm1(self.conv1(x)))
```
Comment (justusschock) [inline deliravision/torch/models/backbones/unet.py]: Does our norm wrapper support no norm at all?

File: tests/test_metrics.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+
+import unittest
+import numpy as np
+
+from deliravision.torch.metrics import dice_score
+
+
+class MetricTest(unittest.TestCase):
```
Comment (justusschock) [inline tests/test_metrics.py]: If moving the numpy metrics to `vision_base`, we need to move this too

File: deliravision/torch/models/basic_networks/classification.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import logging
 from abc import abstractmethod
 import torch
-from delira.models.abstract_network import AbstractPyTorchNetwork
+from delira.models.backends.torch.abstract_network import AbstractPyTorchNetwork
 
 
 class BaseClassificationTorchNetwork(AbstractPyTorchNetwork):
```
Comment (justusschock) [inline deliravision/torch/models/basic_networks/classification.py]: Do we still need this? Or will we provide another way to wrap pre-implemented classifiaction networks?

File: deliravision/torch/models/basic_networks/segmentation.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import logging
 from abc import abstractmethod
 import torch
-from delira.models.abstract_network import AbstractPyTorchNetwork
+from delira.models.backends.torch.abstract_network import AbstractPyTorchNetwork
 
 
 class BaseSegmentationTorchNetwork(AbstractPyTorchNetwork):
```
Comment (justusschock) [inline deliravision/torch/models/basic_networks/segmentation.py]: Do we still need this? Or will we provide another way to wrap pre-implemented segmentation networks?

File: deliravision/torch/metrics.py (hunk 1 above)
Comment (mibaumgartner) [reply deliravision/torch/metrics.py]: Yes, you are right 👍

File: deliravision/torch/models/backbones/unet.py (hunk 4 above)
Comment (mibaumgartner) [reply deliravision/torch/models/backbones/unet.py]: No, currently we do not support  no normalization. But I think we should support it (I just was not shure if I should add this to this pr, but it should be a very minor change)

Comment (justusschock) [reply deliravision/torch/models/backbones/unet.py]: I think you can include it in this PR, since this only is a minor change

File: deliravision/torch/models/basic_networks/classification.py (hunk 6 above)
Comment (mibaumgartner) [reply deliravision/torch/models/basic_networks/classification.py]: I also thought about this, but wasn't shure how what the alternative is. I think, I would prefer that the network has a model instead of subclassing the network for every model. This would decouple them from each, but might be a bit less intuitive. What do you think about this?

Comment (justusschock) [reply deliravision/torch/models/basic_networks/classification.py]: I think I would remove it for now, and reimplement them from scratch witch some rather good default values for both cases. Decoupling them seems nice, but we could also do this via a convert function, which essentially updates the `__dict__` of the `AbstractPyTorchNetwork`. I don't know, which way would be more elegant or straight-forward.

**Comment by justusschock** [comment]
Looks good to me, can you rebase this on master to resolve the conflicts?
